src/METProducer.cc

```diff
@@ -11,7 +11,7 @@ void METProducer::produce(edm::Event& event, const edm::EventSetup& eventSetup)
 
     sumEt = met.sumEt();
 
-    uncorrectedPt = met.uncorrectedPt();
-    uncorrectedPhi = met.uncorrectedPhi();
-    uncorrectedSumEt = met.uncorrectedSumEt();
+    uncorrectedPt = met.uncorPt();
```

Comment:
Indeed. But if ``redoJEC`` is true, then we recreate a new MET, so in this case it should work as long as we don't use the one from MiniAOD